bundles/org.openhab.binding.broadlink/src/main/java/org/openhab/binding/broadlink/internal/handler/BroadlinkBaseThingHandler.java

```diff
@@ -0,0 +1,298 @@
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.util.HexUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Abstract superclass of all supported Broadlink devices.
+ *
+ * @author John Marshall/Cato Sognen - Initial contribution
+ */
+@NonNullByDefault
+public abstract class BroadlinkBaseThingHandler extends BaseThingHandler implements DeviceRediscoveryListener {
+
+    public final Logger logger = LoggerFactory.getLogger(getClass());
+    private static final String INITIAL_DEVICE_ID = "00000000";
+
+    protected BroadlinkDeviceConfiguration thingConfig = new BroadlinkDeviceConfiguration();
+
+    private @Nullable RetryableSocket socket;
+
+    // Can be injected for test purposes
+    private @Nullable NetworkTrafficObserver networkTrafficObserver;
+
+    private int count;
+    private @Nullable ScheduledFuture<?> refreshHandle;
+    private boolean authenticated = false;
+    // These get handed to us by the device after successful authentication:
+    private byte[] deviceId;
+    private byte[] deviceKey;
+
+    public BroadlinkBaseThingHandler(Thing thing) {
+        super(thing);
+        this.deviceId = HexUtils.hexToBytes(INITIAL_DEVICE_ID);
+        this.deviceKey = HexUtils.hexToBytes(BroadlinkBindingConstants.BROADLINK_AUTH_KEY);
+    }
+
+    /**
+     * Method to set the socket manually for test purposes
+     *
+     * @param socket the socket to use
+     */
+    void setSocket(RetryableSocket socket) {
+        this.socket = socket;
+    }
+
+    /**
+     * Method to define a network traffic observer, who can react to the traffic being received.
+     *
+     * @param networkTrafficObserver
+     */
+    void setNetworkTrafficObserver(NetworkTrafficObserver networkTrafficObserver) {
+        this.networkTrafficObserver = networkTrafficObserver;
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        this.thingConfig = getConfigAs(BroadlinkDeviceConfiguration.class);
+        // Validate whether the configuration makes any sense
+        if (thingConfig.isValidConfiguration().length() != 0) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    thingConfig.isValidConfiguration());
+            return;
+        }
+        count = (new Random()).nextInt(65535);
+
+        if (this.socket == null) {
+            this.socket = new RetryableSocket(thingConfig, logger);
+        }
+
+        if (thingConfig.getPollingInterval() != 0) {
+            refreshHandle = scheduler.scheduleWithFixedDelay(new Runnable() {
+                @Override
+                public void run() {
+                    updateItemStatus();
+                }
+            }, 1L, thingConfig.getPollingInterval(), TimeUnit.SECONDS);
```

Comment:
Suggestion might have whitespace issues. Meant as simplification.
```suggestion
            refreshHandle = scheduler.scheduleWithFixedDelay(this::updateItemStatus, 1L,
                    thingConfig.getPollingInterval(), TimeUnit.SECONDS);
```